### PR TITLE
feature(forms): allow custom required indicators for field labels

### PR DESCRIPTION
**Pull request by hypeJunction**
You can now pass required_indicator parameter to elgg_view_input() to override
default indicator view

### CODE REVIEW

**Comment by hypeJunction** [comment]
Motivation here is when we upgrade the login form, we will need to hide the indicator, so we can just pass `'required_indicator' => ''`

File: views/default/elements/forms/label.php [hunk 1]
```diff
@@ -15,10 +16,11 @@
 }
 
 if ($required) {
-	$label .= elgg_format_element('span', [
+	$default_indicator = elgg_format_element('span', [
 		'title' => elgg_echo('field:required'),
 		'class' => 'elgg-required-indicator',
-	], "&ast;");
+			], "&ast;");
+	$label .= elgg_extract('required_indicator', $vars, $default_indicator);
```
Comment (mrclay) [inline views/default/elements/forms/label.php]: I'm not crazy about this pattern (render something then throw it away if not needed). In the past this has led to needless more expensive view renderings. IMO it's also more readable to put all the $vars extractions at the top.

```php
// extractions
...
// set defaults if missing values
...
// render logic
```

**Comment by mrclay** [comment]
Concept is fine. This presents a small BC wrinkle. If a 2.0 plugin has overridden the login form, then 2.1 will cause indicators to appear. Introducing an API in 2.1 is OK, but expecting that API to be used in views is a problem.

**Comment by hypeJunction** [comment]
If login form was overwritten then the form is likely not using elgg_view_input

**Comment by mrclay** [comment]
> If login form was overwritten then the form is likely not using elgg_view_input

:+1:

**Comment by hypeJunction** [comment]
@mrclay Updated. Setting `required_indicator` to an empty string or false, won't render it.

**Pull request merged**